apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -0,0 +1,673 @@
+
+        # Extract Model Library Format tarball.into <project_dir>/model.
+        extract_path = os.path.splitext(project_model_library_format_tar_path)[0]
+        with tarfile.TarFile(project_model_library_format_tar_path) as tf:
+            os.makedirs(extract_path)
+            tf.extractall(path=extract_path)
+
+        if self._is_qemu(options):
+            shutil.copytree(API_SERVER_DIR / "qemu-hack", project_dir / "qemu-hack")
+
+        # Populate CRT.
+        crt_path = project_dir / "crt"
+        crt_path.mkdir()
+        for item in self.CRT_COPY_ITEMS:
+            src_path = os.path.join(standalone_crt_dir, item)
+            dst_path = crt_path / item
+            if os.path.isdir(src_path):
+                shutil.copytree(src_path, dst_path)
+            else:
+                shutil.copy2(src_path, dst_path)
+
+        # Populate Makefile.
+        with open(API_SERVER_DIR / "CMakeLists.txt.template", "r") as cmake_template_f:
+            with open(project_dir / "CMakeLists.txt", "w") as cmake_f:
+                for line in cmake_template_f:
+                    if self.API_SERVER_CRT_LIBS_TOKEN in line:
+                        crt_libs = self.CRT_LIBS_BY_PROJECT_TYPE[options["project_type"]]
+                        line = line.replace("<API_SERVER_CRT_LIBS>", crt_libs)
+
+                    cmake_f.write(line)
+
+        self._create_prj_conf(project_dir, options)
+
+        # Populate crt-config.h
+        crt_config_dir = project_dir / "crt_config"
+        crt_config_dir.mkdir()
+        shutil.copy2(
+            API_SERVER_DIR / "crt_config" / "crt_config.h", crt_config_dir / "crt_config.h"
+        )
+
+        # Populate src/
+        src_dir = project_dir / "src"
+        shutil.copytree(API_SERVER_DIR / "src" / options["project_type"], src_dir)
+
+        # Populate extra_files
+        if options.get("extra_files_tar"):
+            with tarfile.open(options["extra_files_tar"], mode="r:*") as tf:
+                tf.extractall(project_dir)
+
+    def build(self, options):
+        BUILD_DIR.mkdir()
+
+        cmake_args = ["cmake", ".."]
+        if options.get("verbose"):
+            cmake_args.append("-DCMAKE_VERBOSE_MAKEFILE:BOOL=TRUE")
+
+        if options.get("zephyr_base"):
+            cmake_args.append(f"-DZEPHYR_BASE:STRING={options['zephyr_base']}")
+
+        cmake_args.append(f"-DBOARD:STRING={options['zephyr_board']}")
+
+        check_call(cmake_args, cwd=BUILD_DIR)
+
+        args = ["make", "-j2"]
+        if options.get("verbose"):
+            args.append("VERBOSE=1")
+        check_call(args, cwd=BUILD_DIR)
+
+    @classmethod
+    def _is_qemu(cls, options):
+        return (
+            "qemu" in options["zephyr_board"]
+            or
+            # NOTE: mps2_an521 naming breaks the rule that "qemu" is included when qemu is the
+            # launch method.
+            "mps2_an521" in options["zephyr_board"]
```

Comment:
I think it's ok to add a list of boards here instead using the `-qemu` suffix trick as in the old API. However I just would like to mention that there are other boards on Zephyr in the same situation, like the new mps3 boards, and probably there will be more in the future. Zephyr doesn't have a policy to enforce the `qemu-` prefix afaik, unfortunately. That's why I introduced the `-qemu` suffix, so one could add more boards like `mps2_an521` to microTVM without having to update the list here and just using the `-qemu` suffix`, so it would be a bit more generic way to handle such a kind of boards.

Reply:
ah i see. the main reason i went away from it was that the parameter was named `zephyr_board`, so a new user would be confused why they need to add `-qemu` suffix. happy to entertain other parameter names if you have suggestions!